src/EditorFeatures/CSharpTest/Classification/SyntacticTaggerTests.cs

```diff
@@ -35,21 +36,34 @@ public class SyntacticTaggerTests
                     null,
                     new SyntacticClassificationTaggerProvider(null, null, null));
 
-                SnapshotSpan span = default(SnapshotSpan);
+                var expectedLength = subjectBuffer.CurrentSnapshot.Length;
+                int? actualVersionNumber = null;
+                int? actualLength = null;
+                List<string> callstacks = new List<string>();
                 tagComputer.TagsChanged += (s, e) =>
                 {
-                    span = e.Span;
+                    actualVersionNumber = e.Span.Snapshot.Version.VersionNumber;
+                    actualLength = e.Span.Length;
+                    callstacks.Add(StackTrace.GetString());
                     checkpoint.Release();
                 };
 
                 await checkpoint.Task;
+                Assert.Equal(1, actualVersionNumber);
+                Assert.Equal(expectedLength, actualLength);
```

Comment:
Why not add a count to track exactly how many times tags changed is raised?  Essentially

```
Assert.Equal(1, callstacks.Count)
```